src/stream-tcp-inline.c

```diff
@@ -41,9 +41,14 @@ extern int stream_inline;
  *  \retval 0 no
  *  \retval 1 yes
  */
-int StreamTcpInlineMode(void)
+int StreamTcpInlineMode(const Packet *p)
 {
-    return stream_inline;
+    if (stream_inline == PKT_MODE_IDS)
+        return 0;
+    else if (stream_inline == PKT_MODE_IPS)
+        return 1;
+    else /* implied AUTO */
+        return (stream_inline && PacketModeIsIPS(p));
```

Comment:
stream_inline == PKT_MODE_AUTO (stream.inline: auto) means that the value returned depends on the PKT_MODE set in the capture method.
For instance: with nfqueue, this will return 1 (so we're in inline mode), instead 0 with nflog.